code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java

```diff
@@ -0,0 +1,64 @@
+package com.onlinecontacttracing.messaging.filters;
+
+import java.util.ArrayList;
+import com.onlinecontacttracing.messaging.filters.FlaggingFilter;
+import com.onlinecontacttracing.storage.PositiveUser;
+import com.onlinecontacttracing.storage.PotentialContact;
+import java.lang.Exception;
+import com.onlinecontacttracing.messaging.filters.ProfanityList;
+
+//Checks if message contains any profanity
+public class ProfanityFlaggingFilter implements FlaggingFilter{
+  private static ArrayList<String> listOfProfanityIndicators = = new ArrayList<String> () {{
+    add("fuck");
+    add("shit");
+    add("bitch");
+    add("crap");
+    add("damn");
+    add("goddamn");
```

Comment:
This is mostly fine, but most likely you'd actually define some dictionary file you read in. This makes it much easier to expand upon.